doc/source/whatsnew/v0.24.0.rst

```diff
@@ -1560,6 +1560,7 @@ Groupby/Resample/Rolling
 - Bug in :meth:`pandas.core.groupby.GroupBy.rank` with ``method='dense'`` and ``pct=True`` when a group has only one member would raise a ``ZeroDivisionError`` (:issue:`23666`).
 - Calling :meth:`pandas.core.groupby.GroupBy.rank` with empty groups and ``pct=True`` was raising a ``ZeroDivisionError`` (:issue:`22519`)
 - Bug in :meth:`DataFrame.resample` when resampling ``NaT`` in ``TimeDeltaIndex`` (:issue:`13223`).
+- Bug in :meth:`DataFrame.groupby` did not work correctly with ``observed=True`` when aggregating a specified column (:issue:`23970`)
```

Comment:
this is when its a categorical column; also this is only for ``as_index=False``, right?

Reply:
Any operation over a selected column was effectively ignoring the ``observed`` parameter that was being passed to it. With ``as_index=True`` the `observed` value isn't used so the code behaves as expected.